beacon-chain/blockchain/process_block.go

```diff
@@ -514,12 +515,35 @@ func missingIndices(bs *filesystem.BlobStorage, root [32]byte, expected [][]byte
 	return missing, nil
 }
 
+func missingDataColumns(bs *filesystem.BlobStorage, root [32]byte, expected map[uint64]bool) (map[uint64]struct{}, error) {
+	if len(expected) == 0 {
+		return nil, nil
+	}
+	if len(expected) > int(params.BeaconConfig().NumberOfColumns) {
+		return nil, errMaxDataColumnsExceeded
+	}
+	indices, err := bs.Indices(root)
+	if err != nil {
+		return nil, err
+	}
+	missing := make(map[uint64]struct{}, len(expected))
+	for col := range expected {
+		if !indices[col] {
+			missing[col] = struct{}{}
+		}
+	}
+	return missing, nil
+}
```

Comment:
```suggestion
func missingDataColumns(bs *filesystem.BlobStorage, root [32]byte, expected map[uint64]bool) (map[uint64]bool, error) {
	if len(expected) == 0 {
		return nil, nil
	}
	if len(expected) > int(params.BeaconConfig().NumberOfColumns) {
		return nil, errMaxDataColumnsExceeded
	}
	indices, err := bs.Indices(root)
	if err != nil {
		return nil, err
	}
	missing := make(map[uint64]bool, len(expected))
	for col := range expected {
		if !indices[col] {
			missing[col] = true
		}
	}
	return missing, nil
}
```

Reply:
Advantages of `map[uint64]bool` over `map[uint64]struct{}`:

You don't need the (a little bit ugly) `missing[col] = struct{}{}`
It's very convenient to check if and item is in the set:

```go
if mySet[item] {
    ...
}
```

Reply:
I can do this but would necessitate that columns and blobs have separate DA methods in that case

Reply:
Hum normally just accepting the suggestion without changing anything else should work.

Reply:
Was referring to this comment:
https://github.com/prysmaticlabs/prysm/pull/13938#discussion_r1584443104